### DISCUSSION TITLE
How to tell Renovate to delete all the existing PRs and branches created by Renovate?

### DISCUSSION THREAD

**Discussion by phaninra** [renovatebot/renovate#17930] [Request Help] [answered]
How to tell Renovate to delete all the existing open PRs and branches created by Renovate? Is there a way to do this?  

This might come in handy in various accidental or otherwise normal scenarios. For example, you want to cleanup all the PRs opened by Renovate because you forgot to uncomment the `*` package grouping, etc.

**Comment by viceice** [comment, marked as answer]
renovate will auto close non required PR if no longer needed, it will also delete obsolete branches.

**Reply by phaninra** [reply]
If I understand correctly, you're referring to the 'happy path' non-required PRs, i.e. the dependency has already been updated separately, etc. 

I was referring to potential accidents and ways to immediately cleanup. For example, I was experimenting with something. I ran the Renovate without the `matchPackagePatterns": ["*"]` which created 100+ PRs across several internal repositories. The only way to cleanup was to visit all those repos and delete them manually. Please note that some Git hosts do not offer a "bulk select" option in UI for PR deletion. I wish there's a way to tell Renovate to delete all existing PRs, maybe supplemented by a time based filter on PR creation time.

**Reply by rarkins** [reply]
I'm not sure you understand correctly. Renovate by default will always auto close PRs which it thinks shouldn't exist. So if you ran it a next time on the same repos with a fixed config then you should see mass auto closing.

**Reply by phaninra** [reply]
Thank you for elaborating on that. Let me try that out.

**Reply by phaninra** [reply]
Thanks again. It is working as you described - auto-closes open PRs which don't match the config from current run. 

On a side note, is there a way to tell Renovate to not autoclose the MRs? 

For example, we have a requirement to upgrade some stuff every x weeks and other stuff every y weeks. Being new to Renovate, I thought I could create 2 separate config.js files to capture the rules for each run. Now I realized that every run driven by config1.js would automatically close the open MRs created by the config2.js. How can I avoid that?

**Reply by viceice** [reply]
checkout the schedule config option in the docs

**Reply by rarkins** [reply]
pruneStaleBranches=false

**Reply by phaninra** [reply]
(Please let me know if I should rather open a new discussion)

 Thanks for pointing that out. The schedule feature addresses some scenarios. Here's something else ..

```
    //config1.js
    gradle: {
        includePaths: ["**/path1"]
    }
    packageRules: [...]
```
```
    //config2.js
    gradle: {
        includePaths: ["**/path2"]
    }
   packageRules: [...]
```

1) Combining both into a single file can be challenging or complicated because of overlaps among the rules. Note that they are for the same package manager.  I think a `Don't autoclose PRs`  feature, if exists, would be handy.

2) Will there be a way to run for only the "path1" at will, i.e. with a manual kickoff ?  (No scheduling constraints involved)

**Reply by phaninra** [reply]
@rarkins  `pruneStaleBranches=false` works. Thank you.
